esmvalcore/config/_config_object.py

```diff
@@ -93,29 +112,140 @@ def _load_default_config(cls, filename: Union[os.PathLike, str]):
 
         return new
 
+    @staticmethod
+    def _read_config_file(config_user_path: Path) -> dict:
+        """Read configuration file and store settings in a dictionary."""
+        if not config_user_path.is_file():
+            raise FileNotFoundError(
+                f"Config file '{config_user_path}' does not exist"
+            )
+
+        with open(config_user_path, 'r', encoding='utf-8') as file:
+            cfg = yaml.safe_load(file)
+
+        return cfg
+
+    @staticmethod
+    def _get_config_user_path(
+        filename: Optional[os.PathLike | str] = None
+    ) -> Path:
+        """Get path to user configuration file.
+
+        `filename` can be given as absolute path, or as path relative to the
+        current working directory or `~/.esmvaltool`.
+
+        If `filename` is not given, try to get user configuration file from the
+        following locations (sorted by descending priority):
+
+        1. Internal `_ESMVALTOOL_USER_CONFIG_FILE_` environment variable
+        (this ensures that any subprocess spawned by the esmvaltool program
+        will use the correct user configuration file).
+        2. Command line arguments `--config-file` or `--config_file` (only if
+        script name is `esmvaltool`).
+        3. `config-user.yml` within default ESMValTool configuration directory
+        `~/.esmvaltool`.
+
+        Note
+        ----
+        This will NOT check if the returned file actually exists to allow
+        loading the module without any configuration file (this is relevant if
+        the module is used within a script or notebook). To check if the file
+        actually exists, use the method `load_from_file` (this is done when
+        using the `esmvaltool` CLI).
+
+        If used within the esmvaltool program, set the
+        _ESMVALTOOL_USER_CONFIG_FILE_ at the end of this method to make sure
+        that subsequent calls of this method (also in suprocesses) use the
+        correct user configuration file.
+
+        """
+        # (1) Try to get user configuration file from `filename` argument (also
+        # test relative to ~/.esmvaltool)
+        config_user = filename
+
+        # (2) Try to get user configuration file from internal
+        # _ESMVALTOOL_USER_CONFIG_FILE_ environment variable
+        if (
+                config_user is None and
+                '_ESMVALTOOL_USER_CONFIG_FILE_' in os.environ
+        ):
+            config_user = os.environ['_ESMVALTOOL_USER_CONFIG_FILE_']
+
+        # (3) Try to get user configuration file from CLI arguments
+        if config_user is None:
+            config_user = Config._get_config_path_from_cli()
+
+        # (4) Default location
+        if config_user is None:
```

Comment:
will `Config._get_config_path_from_cli()` return a None if that's not specified via cmd line? Or an empty string?

Reply:
It will return `None`, so we are fine :+1: